### DISCUSSION TITLE
Support TARGET_TEMPERATURE_RANGE for water heater entities

### DISCUSSION THREAD

**Discussion by kamaradclimber** [home-assistant/architecture#995] [Entity Models] [answered]
Current status
---

Water heater entities expose [target_temperature_high](https://developers.home-assistant.io/docs/core/entity/water-heater) and target_temperature_low but those properties are not exposed via the UI.
Climate entities support two distinct features: ability to set target temperature or to set target temperature range.

Goal
----

Water heater with the ability to set a range of target temperature should be able to be represented as such:
- with a service call to update the range
- with the UI to show the range
- with the UI to allow to update the the range

Related links
---

- https://github.com/home-assistant/architecture/discussions/923
- https://github.com/kamaradclimber/heishamon-homeassistant/issues/145
- https://github.com/home-assistant/architecture/discussions/518

**Comment by emontnemery** [comment]
Link to core PR: https://github.com/home-assistant/core/pull/102960

**Comment by MartinHjelmare** [comment]
How many of the existing integrations that have a water heater platform in Home Assistant could leverage a target temperature range? We normally want a handful of integrations that can use a feature before we add it in Home Assistant.

Please list the integrations that you find.

**Comment by geduxas** [comment]
i think it will be almost all supported heatpumps, also maybe some additional smart boiler's :)

**Reply by MartinHjelmare** [reply]
Please list the integrations explicitly that you have checked.

**Reply by geduxas** [reply]
Oh my :D i thought it will be easy task, listed known big heatpump names in official integration's and there is only NYBE.. others are only AC... So most of them i think will be custom integrations or from HACS... Sorry for my interruption here, i just waiting too for better control ability of heatpump's

**Comment by kamaradclimber** [comment]
There are currently 16 integrations in HA core exposing a water heater entity (including mqtt integration).
3 of those (mqtt, osoenergy and nibe_heatpump) set the `target_temperature_high` and `target_temperature_low` properties which means they would benefit from a target_range service.

I don't know how to count custom integrations in the ecosystems, I can only quote the integration I maintain (https://github.com/kamaradclimber/heishamon-homeassistant) which would benefit from it as well.

For reference, the feature exists in climate entities (110 core integrations expose a climate entity, 25 benefit from the service because they expose temperature range).

**Comment by emontnemery** [comment]
@kamaradclimber Is there a reason why the temperature range should be a property of the water heater entity itself instead of being exposed as a separate entity, for example a number entity named hysteresis?

**Reply by kamaradclimber** [reply]
Hello, I’m not sure to understand the proposal.

Target temperature range is an essential property of the water heater (as is the current temperature or the target temperature). How would you represent the heater if it was an independent entity?

**Reply by emontnemery** [reply]
Is the hysteresis to control when the heating element is turned on really such an essential property of a water heater that it adds value to show it in the UI, or does it just add clutter?

Just to be clear: I'm not saying the target temperature range does not add value, but I don't see a clear motivation in this discussion explaining why it's needed.

**Reply by kamaradclimber** [reply]
Fair point!
My opinion is that range/hysteresis provides value:
-  To explain why heater is not heating when water temperature below what people traditionally associate with "target temperature" 
-  Most installers/manual/online resource recommend to set target temperature to xx°C for health issue but often forget to precise most waters heaters have hysteresis control (not always configurable though), often with large range. People setting target temperature based on a perceived "healthy" temperature but do not account for hysteresis.

Regarding cluttering the interface, it’s a valid point. I think the current UI (showing a gauge on a ring) can easily adds a second reference point. Gauges often have delimited areas (for instance https://community.grafana.com/t/same-size-gauge/80167).

Finally, I think it’s important to surface the hysteresis control, when it exists, to favor energy savings which is one of the  common motivation for users of HA.

**Comment by frenck** [comment, marked as answer]
We have discussed this feature, and agreed to sync this up with the climate platform.

The implementation can thus go ahead. Please note: This needs frontend changes too.

../Frenck
